ext/php8/configuration.h

```diff
@@ -4,10 +4,38 @@
 #include <stdbool.h>
 
 #include "compatibility.h"
+#include "ddtrace_config.h"
 #include "ddtrace_string.h"
 #include "env_config.h"
 #include "integrations/integrations.h"
 
+// TODO Tie these into the X Macros below
+static inline bool get_dd_trace_debug(void) {
+    zval *val = zai_config_get_value(DDTRACE_CONFIG_DD_TRACE_DEBUG);
+    return Z_TYPE_P(val) == IS_TRUE;
+}
+
+static inline zai_string_view get_dd_service(void) {
+    zval *val = zai_config_get_value(DDTRACE_CONFIG_DD_SERVICE);
+    return (zai_string_view){.len = Z_STRLEN_P(val), .ptr = Z_STRVAL_P(val)};
+}
+
+static inline zend_array *get_dd_tags(void) {
+    zval *val = zai_config_get_value(DDTRACE_CONFIG_DD_TAGS);
+    GC_ADDREF(Z_ARRVAL_P(val));
+    return Z_ARRVAL_P(val);
+}
+
+static inline int64_t get_dd_trace_agent_port(void) {
+    zval *val = zai_config_get_value(DDTRACE_CONFIG_DD_TRACE_AGENT_PORT);
+    return (int64_t)Z_LVAL_P(val);
```

Comment:
We probably should just use `zend_long` here, i.e. the native data type. I wouldn't really say something is a 64 bit integer when it actually will only ever use 32 bits.

Reply:
For this specific case that targets the Agent port, I agree on using 32 bit ints (or smaller). Although this getter is specific to the Agent port, I was planning on incorporating these functions into the X Macro setup we have for the existing config. In that case, I'm not sure we want to artificially limit all `int` configuration values to 32-bit.

Also, since `zend_long` is a feature of PHP 7+, what do you suggest for PHP 5? Do you see any value in providing a generic type in this case to reduce the minor API changes across the major PHP versions?

Reply:
php5 uses simply a long. Okay, I agree, isn't going to do any harm to always use int64_t.